src/main/java/com/swp/board/domain/BoardService.java

```diff
@@ -0,0 +1,120 @@
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@RequiredArgsConstructor
+@Service
+public class BoardService {
+
+    private final BoardRepository boardRepository;
+    private final UserRepository userRepository;
+    private final StudyRepository studyRepository;
+
+    @Transactional
+    public void createBoard(JwtUserDetails userDetails, BoardCreateRequestDto boardCreateRequestDto) {
+        User user = userRepository.findByProviderAndProviderId(userDetails.getProvider(), userDetails.getUsername())
+                .orElseThrow(() -> new UserNotFoundException("없는 유저입니다"));
+
+        Study study = studyRepository.findById(boardCreateRequestDto.getStudyId())
+                .orElseThrow(() -> new StudyNotFoundException("공부 정보를 찾을 수 없습니다."));
+
+        boardRepository.save(Board.builder()
+                .user(user)
+                .study(study)
+                .content(boardCreateRequestDto.getContent())
+                .build());
+    }
+
+    @Transactional
+    public void updateBoard(JwtUserDetails userDetails, BoardUpdateRequestDto boardUpdateRequestDto) {
+        Board board = boardRepository.findById(boardUpdateRequestDto.getBoardId())
+                .orElseThrow(() -> new BoardNotFoundException("글을 찾을 수 없습니다."));
+
+        User user = userRepository.findByProviderAndProviderId(userDetails.getProvider(), userDetails.getUsername())
+                .orElseThrow(() -> new UserNotFoundException("없는 유저입니다"));
+
+        if (user.equals(board.getUser())) {
+            //글을 쓴 사람만이 글을 수정할 권한이 있다고 가정함
+            board.updateContent(boardUpdateRequestDto.getContent());
+            boardRepository.save(board);
+        } else {
+            throw new BoardForbiddenException("글을 수정할 권한이 없습니다.");
+        }
+    }
+
+    @Transactional
+    public void deleteBoard(JwtUserDetails userDetails, BoardDeleteRequestDto boardDeleteRequestDto) {
+        Board board = boardRepository.findById(boardDeleteRequestDto.getBoardId())
+                .orElseThrow(() -> new BoardNotFoundException("글을 찾을 수 없습니다."));
+
+        User user = userRepository.findByProviderAndProviderId(userDetails.getProvider(), userDetails.getUsername())
+                .orElseThrow(() -> new UserNotFoundException("없는 유저입니다"));
+
+        if (user.getRole().equals(Role.ADMIN) || user.equals(board.getUser())) {
+            //Admin, 글을 쓴 사람만이 글을 삭제할 권한이 있다고 가정함
+            boardRepository.delete(board);
+        } else {
+            throw new BoardForbiddenException("글을 수정할 권한이 없습니다.");
+        }
+    }
+
+    @Transactional
+    public BoardResponseDto getBoard(Integer boardId) {
+        Board board = boardRepository.findById(boardId)
+                .orElseThrow(() -> new BoardNotFoundException("글을 찾을 수 없습니다."));
+
+        return BoardResponseDto.builder()
+                .nickname(board.getUser().getNickname())
+                .content(board.getContent())
+                .createdAt(board.getCreatedAt())
+                .build();
+    }
```

Comment:
아래 stream에서도 그렇고 `BoardResponseDto.builder()...build()`가 반복되는데, 이를 `BoardResponseDto`에 public static method로 만들어서 아래처럼 간략하게 사용하도록 하면 어떨까요?

```suggestion
        return BoardResponseDto.from(board);
    }
```

Reply:
@sang-w0o 오 저도 이게 좋을 것 같습니다. `.of()`는 어떨까요?

Reply:
좋은 것 같습니다. 그렇게 수정할게요

Reply:
`of()`는 `LocalDateTime`을 예로 들자면 새로운 객체를 만들어내기 위해 필요한 **값** 들을 인자로 전달받아 만들어주는데, 이 경우에는 다른 **객체** *로부터* 만들어지는 거라 `from()`이 낫다고 생각해요. 물론 주관적인 생각입니당

Reply:
다시 from으로 바꿀게요 그러면